feat(APP-381): Enhance DAO settings page with SubDAOs

tyhonchik opened the pull request:
# Description

- Improved layout.
- Added breadcrumbs.

## Type of Change

<!--- Please delete the options that are not relevant. -->

- [ ] Major: Breaking change (change that would cause existing functionality to not work as expected)
- [ ] Minor: Feature (non-breaking change which adds new functionality)
- [x] Patch: Enhancement (non-breaking change to an existing feature)
- [ ] Patch: Bug fix (non-breaking change which fixes an issue)

## Developer Checklist:

- [x] Manually smoke tested the functionality in a preview or locally
- [x] Confirmed there are no new warnings or errors in the browser console
- [ ] (For User Stories only) Double-checked that all Acceptance Criteria are satisfied
- [x] Confirmed there are no new warnings on automated tests
- [x] Merged and published any dependent changes in downstream modules
- [x] Selected the correct base branch
- [x] Commented the code in hard-to-understand areas
- [x] Followed the code style guidelines of this project
- [x] Reviewed that the Files Changed in Github’s UI reflect my intended changes
- [ ] Confirmed the pipeline checks are not failing

## Review Checklist:

- [ ] (For User Stories only) Tested in a preview or locally that all Acceptance Criteria are satisfied
- [ ] Confirmed that changes follow the code style guidelines of this project

jamesej reviewed (approved):
Looks good, thanks

tyhonchik committed:
feat(APP-381): Enhance DAO settings page with SubDAOs:

- Improved layout.
- Added breadcrumbs.

thekidnamedkd commented on .changeset/vast-moons-lose.md:
```diff
@@ -0,0 +1,5 @@
+'@aragon/app': patch
+---
+
+Improve the settings page according to the designs in Figma.
```
```suggestion
Improve the settings page according to the designs in Figma
```

thekidnamedkd commented on src/assets/locales/en.json:
```diff
@@ -2671,14 +2671,17 @@
         "chain": "Chain",
-        "ens": "ENS name",
-        "links": "Resources",
+        "ens": "ENS Name",
```
I don't agree but we actually have a pattern of not capitalizing all words in a title/label/button etc
```suggestion
        "ens": "ENS name",
```

tyhonchik replied:
I changed it because of figma https://www.figma.com/design/LFe79BV9DoC1VB96FvjT7d/APP-278--Capital-flow-primitives?node-id=17571-24779&m=dev

thekidnamedkd commented on src/modules/settings/components/daoHierarchy/daoHierarchy.tsx:
```diff
@@ -25,60 +27,84 @@ export interface IDaoHierarchyProps {
     currentDaoId: string;
 }
 
-const DaoInfo: React.FC<{ dao: IDao | ISubDaoSummary }> = ({ dao }) => {
+interface IDaoInfoProps {
+    dao: IDao | ISubDaoSummary;
+    isMainDao: boolean;
+}
```
JS Docs

thekidnamedkd commented on src/modules/settings/components/daoHierarchy/daoHierarchy.tsx:
```diff
@@ -101,31 +127,43 @@ export const DaoHierarchy: React.FC<IDaoHierarchyProps> = (props) => {
-                        <div className="flex items-center gap-2">
-                            <p className="font-medium">{dao.name}</p>
-                            <Tag label={t('app.settings.daoHierarchy.mainDaoLabel')} />
+                    <Accordion.ItemHeader className="!items-center">
```
items-center !important? what's the reasoning? if this is an alignment issue in general we should fix it on the govkit component.

thekidnamedkd commented on src/shared/components/processDataListItem/daoBreadcrumbs.tsx:
```diff
@@ -0,0 +1,101 @@
+    return dfs(buildHierarchyNode(dao));
+};
+
+interface IDaoBreadcrumbsProps {
```
JS Docs

thekidnamedkd commented on src/shared/components/processDataListItem/daoBreadcrumbs.tsx:
```diff
@@ -0,0 +1,101 @@
+type HierarchyNode = Pick<IDao, 'address' | 'name' | 'avatar'> & {
+    subDaos?: Array<ISubDaoSummary & { subDaos?: ISubDaoSummary[] }>;
+};
+
```
Docs

thekidnamedkd commented on src/shared/components/processDataListItem/daoBreadcrumbs.tsx:
```diff
@@ -0,0 +1,101 @@
+ * Depth-first search for the path from the root DAO to the target address.
+ * Returns the breadcrumb chain or undefined when not found.
+ */
+const findDaoPath = (params: { dao?: IDao; targetAddress?: string }): BreadcrumbNode[] | undefined => {
```
Seems like these could be a utils with the usual pattern.

tyhonchik replied:
extracted to daoBreadcrumbsUtils

thekidnamedkd reviewed (commented):
Nice. Left a couple annoying suggestions. 🙃

tyhonchik committed:
feat(APP-381): Fix PR issues

- Separate component for DaoBreadcrumbs
- Separate utils for building breadcrumbs

tyhonchik commented:
@thekidnamedkd thanks for the review
addressed all issues

thekidnamedkd commented on src/shared/components/daoBreadcrumbs/daoBreadcrumbs.tsx:
```diff
@@ -0,0 +1,46 @@
+    path?: BreadcrumbNode[];
+}
+
+/**
+ * Simple component for displaying a DAO hierarchy as breadcrumbs.
+ *
+ * The actual path resolution is delegated to `daoBreadcrumbsUtils.buildDaoBreadcrumbPath`.
+ */
```
Minor, but I don't think we need docs for UI components

tyhonchik replied:
I left it just for keeping a reference for a daoBreadcrumbsUtils utility to collect breadcrumbs

Overall, I agree - the components speak for themselves

thekidnamedkd reviewed (approved):
LGTM (left one tiny comment) 👍
